examples/envelope_designer/sketch.js

```diff
@@ -0,0 +1,103 @@
+var attackTime = 0.001
+var attackLevel = 1.0;
+var decayTime = 0.2;
+var decayLevel = 0.2;
+var releaseTime = 0.5;
+var releaseLevel = 0;
+var sustainTime = 0.5;
+
+var attackTimeInput, attackLevelInput;
+var decayTimeInput, decayLevelInput;
+var releaseTimeInput, releaseLevelInput;
+var sustainTimeInput;
+
+var maxTime = 1;
+var maxLevel = 5;
+var xScale = 100;
+
+var env, osc;
+
+function setup() {
+  var cnv = createCanvas(500, 500);
+  cnv.mousePressed(playEnv);
+
+  env = new p5.Env();  
+  osc = new p5.Oscillator('triangle');
+  osc.amp(env);
+  osc.start();
+  osc.freq(440);
+
+  attackTimeInput = createSlider(0, maxTime, attackTime, 0);
+  attackTimeInput.position(20, 15);
+  attackLevelInput = createSlider(0, maxLevel, attackLevel, 0);
+  attackLevelInput.position(160, 15);
+  decayTimeInput = createSlider(0, maxTime, decayTime, 0);
+  decayTimeInput.position(20, 45);
+  decayLevelInput = createSlider(0, maxLevel, decayLevel, 0);
+  decayLevelInput.position(160, 45);
+  sustainTimeInput = createSlider(0, maxTime, sustainTime, 0);
+  sustainTimeInput.position(20, 75);
+  releaseTimeInput = createSlider(0, maxTime, releaseTime, 0);
+  releaseTimeInput.position(20, 105);
+  releaseLevelInput = createSlider(0, maxLevel, releaseLevel, 0);
+  releaseLevelInput.position(160, 105);
+}
+
+function draw() {
+  background(30, 100, 155);
+  attackTime = attackTimeInput.value();
+  attackLevel = attackLevelInput.value();
+  decayTime = decayTimeInput.value();
+  decayLevel = decayLevelInput.value();
+  sustainTime = sustainTimeInput.value();
+  releaseTime = releaseTimeInput.value();
+  releaseLevel = releaseLevelInput.value();
+  drawADSR();
+
+  // Text
+  textAlign(CENTER);
+  text('click to play', width/2, height/2);
+}
+
+function playEnv() {
+  env.set(attackTime, attackLevel, decayTime, decayLevel, releaseTime, releaseLevel);
+  env.play(osc, 0, sustainTime);
```

Comment:
A bit of a tangent, but I thought it might be interesting to see these methods as part of the example.

And then, I thought of this classic essay by Bret Victor
http://worrydream.com/LearnableProgramming/

(which may have inspired [Processing Tweak Mode](http://galsasson.com/tweakmode/))

...could be good inspiration as you create more examples and think about the user experience from both a code and UI perspective.

Reply:
Wow, that essay was eye-opening, and very inspiring! I do feel a definite need to illustrate better how all the different sound components work, but I'm not sure that tacking more onto this example is the best way to do achieve that. 

Let's talk about this more in terms of designing the core introduction-to-p5.js-sound tutorials. :)